TRex2-lib/src/Packets/RulePkt.h

```diff
@@ -193,13 +193,13 @@ class RulePkt {
   /**
    * Fills leaves with the set of indexes that are leaves in the ordering graph
    */
-  void getLeaves(std::set<int>& leaves);
+  std::set<int> getLeaves();
```

Comment:
Return collections by value, because with c++11 `move semantic` and compilers `RVO` it is no longer considered an expensive operation.